### PR TITLE
CP-41279: add migration_compression pool option

### PR DESCRIPTION
**Pull request by snwoods**
Signed-off-by: Steven Woods <steven.woods@citrix.com>

### CODE REVIEW

File: ocaml/idl/datamodel_pool.ml [hunk 1]
```diff
@@ -1322,6 +1322,10 @@ let t =
             ~default_value:(Some (VRef null_ref)) "repository_proxy_password"
             "Password for the authentication of the proxy used in syncing with \
              the enabled repositories"
+        ; field ~lifecycle:[] ~ty:Bool ~default_value:(Some (VBool false))
```
Comment (robhoes) [inline ocaml/idl/datamodel_pool.ml:1325]: You should add `~qualifier:DynamicRO`. I'm not sure why the compiler didn't complain about this.

File: ocaml/idl/datamodel_lifecycle.ml [hunk 2]
```diff
@@ -19,6 +19,8 @@ let prototyped_of_field = function
       Some "22.27.0"
   | "host", "last_software_update" ->
       Some "22.20.0"
+  | "pool", "migration_compression" ->
+      Some "22.32.0"
```
Comment (robhoes) [inline ocaml/idl/datamodel_lifecycle.ml:23]: This should have `22.32.0-next` if the autogenerator works properly...

Comment (snwoods) [reply ocaml/idl/datamodel_lifecycle.ml:23]: It gave me 22.32.0-next the first time but after some changes only gave me the above

**Comment by robhoes** [comment]
Looks like you need to update the unit tests.

**Review by lindig — APPROVED** [review]

File: ocaml/idl/datamodel_pool.ml (hunk 1 above)
Comment (snwoods) [reply ocaml/idl/datamodel_pool.ml]: Thanks, although it should be RW right?

**Review by robhoes — APPROVED** [review]

**Pull request merged**